src/marqo/s2_inference/clip_utils.py

```diff
@@ -129,6 +129,20 @@ def load_image_from_path(image_path: str, image_download_headers: dict, timeout_
     return img
 
 
+def validate_url(url: str) -> bool:
+    """Validate a URL to ensure it is a valid URL. Returns True if the URL is valid or the encoded URL is valid.
+    Args:
+        url (str): URL to validate.
+    Returns:
+        bool: True if the URL is valid, False otherwise.
+    """
+    if isinstance(url, str):
+        return validators.url(url) or validators.url(encode_url(url))
```

Comment:
What are we trying to cover with encode_url? Is there a risk of regression or breaking change?

Reply:
A customer URL that is not recognized as a valid URL. I think this is the expected behaviour because we are encoding the URL when downloading the image with a URL. So we should apply the same encoding when validating the URL.